File: 4. Median of Two Sorted Arrays/Simpleton Approach/median_of_two_sorted_arrays.js
function toInteger (number){
	return (number - (number % 1));
}

function getMedian (list){
	let lenList = list.length;
	let evenLength = (lenList % 2 == 0);
	let median = 0.0;
	
	if (evenLength){
		median = (list[toInteger(lenList / 2)] + list[toInteger(lenList / 2) - 1]) / 2; 
	} else {
		median = (list[toInteger(lenList / 2)]);
	}
	
	return (median);
}

function combineList (list1, list2){
	let combinedList = [];
	
	for (i of list1){
		combinedList.push(i);
	}
	
	for (i of list2){
		combinedList.push(i);
	}
	
	combinedList.sort(function(a, b) {return a - b})
	
	return (combinedList);
}

let list1 = [1,3,5];
let list2 = [2,4];

let list_combined = combineList(list1, list2)

let median = getMedian(list_combined);

console.log(median);
console.log("\n\n\n");

File: 4. Median of Two Sorted Arrays/Simpleton Approach/median_of_two_sorted_arrays.kt
fun get_median (list: Array<Int>): Float{
	val len_list: Int = list.size;
	val even_length: Boolean = (len_list % 2 == 0);
	var median: Float = 0.0f;
	
	if (even_length){
		median = ((list[len_list / 2].toFloat() + list[(len_list / 2) - 1].toFloat()) / 2);
	} else {
		median = (list[len_list / 2]).toFloat();
	}
	
	return (median);
}

fun combine_list (list1: Array<Int>, list2: Array<Int>): Array<Int>{
	val len_list1: Int = list1.size;
	val len_list2: Int = list2.size;
	var combined_list: Array<Int> = Array(len_list1 + len_list2) {0};
	var combined_list_cursor: Int = 0;
	for (i in 0 until len_list1){
		combined_list[combined_list_cursor] = list1[i];
		combined_list_cursor += 1;
	}
	
	for (i in 0 until len_list2){
		combined_list[combined_list_cursor] = list2[i];
		combined_list_cursor += 1;
	}
	
	//sort
	combined_list.sort();
	
	return (combined_list);
}

fun main (){
	val list1: Array<Int> = arrayOf(1,3);
	val list2: Array<Int> = arrayOf(2,4);
	
	var list_combined = combine_list(list1, list2);
	
	var median = get_median(list_combined);
	
	println(median);
	println("\n\n\n");
}
